[I Muthukumar] iP

syncode98 opened the pull request.

syncode98 committed:
Commenting the "DUKE" symbol

syncode98 committed:
minimum requirements for first task without the lines

syncode98 committed:
edited unnecessary comments from the program and added the lines

syncode98 committed:
final edit

syncode98 committed:
clean code

syncode98 committed:
Implementing level-0 for the code

The Duke symbol was removed from the code.
This symbol was used to test out the code when it was cloned from
a remote directory.

Two statements and a few lines were added.These act as a base
for future communications.

This the is required task for week 2.

syncode98 committed:
Implement Level 1 for Chatbot

Chatbot echo's user's commands and exits when
the user types "bye".

Function printLines prints dotted lines. This improves
 code readability.

Users would be able to enter their name and Chatbot
records it. Addresses the user with the name
given.

syncode98 committed:
Implement Level-2

Chatbot prompts user to add in tasks
and displays it to the user when the command
"list" is used.

Class Task keeps track of the
String array that stores all the tasks. The
class contains the relevant constructors, getters
and setters for the class.

Function addTask in Task.java adds the task
to the array and function printTasks prints the existing
tasks if user enters the command "list".

Initialise function in Duke .java prompts
user to enter name

syncode98 committed:
Implement Level 3

The chatbot is able to update the completion statuses of the tasks that
were given by the user. The command "done " prompts the system to
change the completion status of the task as completed.

In the Task Class, the completion array stores the completion statuses of the tasks. The
completeTasks function controls the completion statuses
of the tasks.The appropriate getters and setters have been added for this class
level member

The printTasks function now accommodates the printing of
the tick and cross symbols for the task.

In the Duke Class, the commands function reads the input of the users
and decides the appropriate course of action.

syncode98 committed:
Implement A-CodingStandard

Function command () causes confusion to readers.
Hence, it is changed to readInput to comply with the Coding
Standards.

Implentation of the task class as an array of objects
makes it harder for the tasks to be modified in the future. Hence, the class
now initialises single task objects. The array is then initialised
in the main function.

The functions in the Task class now modify the specific
task it is called upon to modify.

syncode98 committed:
Implement Level 4

Classes Todo,Deadline and Event were created to
track the different types of tasks that are available. There
classes inherit from the Task class.

Funciton toString has been defined in all of the classes
(except Duke class) .The function returns the string representation
of the class. Furthermore, this function overrides the original
toString function in Object class.

The seperate task classes(Todo,Deadline and Event) each have overridden
 toString methods,which add on the specific character of the type of  task
to be printed out.

syncode98 committed:
Implemened Automated Text UI Testing

Main file able to read inputs from the input.txt
file and pass the outputs to the ACTUAL.txt file.
After that, the ACTUAL.txt file is compared with the
EXPECTED.txt file to examine any differences.

syncode98 committed:
Display Unicode characters

Runtest.bat file modified so that the EXPECTED.txt
now displays the unicode characters. Previously,
it only displayed question marks.

jazhten commented on src/main/java/Duke.java:
```diff
@@ -15,41 +26,31 @@ public static void initiliase(){
+    public static void readInput(Task[] tasks) {
         Scanner inputScanner= new Scanner(System.in);
         String command;
         command=inputScanner.nextLine();
```
You might consider leaving a space before and after the "=" to improve the readability.

syncode98 replied:
Will take note of that.

jazhten commented on src/main/java/Duke.java:
```diff
@@ -1,11 +1,22 @@
+
+    public static void main(String[] args) {
+        Task[] tasks = new Task[MAX_SIZE];
+        for (int i=0; i<MAX_SIZE; i++) {
```
Perhaps you could leave some space in the loop forming stage, for example i < MAX_SIZE; to improve readability.

syncode98 replied:
Will look into that in my subsequent commits. Thank you for your comments!

jazhten reviewed (commented):
Nice code and I like how you added a personalisation touch by asking for the user's name. You might want to space out the lines a little to make the code slightly easier to read.

EdmundEXE commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,80 @@
 public class Duke {
+
+    public static final int MAX_SIZE = 100;
+    private static Task[] tasks=new Task[MAX_SIZE];
+    public static Scanner inputScanner= new Scanner(System.in);
```
Consider the spacing between  '=' consistently

EdmundEXE commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,15 @@
+        super(task, isDone);
+    }
+
+    public Todo(String task) {
+        super(task,false);
```
Consider the spacing between ','. eg ( blabla(abc, xyz) )

EdmundEXE commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,73 @@
+public class Task {
+    private String task;
+    private boolean isDone;
```
variable is in camelCase and sound 'boolean'. Good!

syncode98 replied:
Thank you!

EdmundEXE commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,80 @@
+                addToTasks(todo);
+
+            } else if (command.contains("deadline")) {
+                String[] words=command.split("/");
```
Plural form name used for a collection of objects. Nice!

syncode98 replied:
Thank you!

syncode98 committed:
Initialise String constants

In the Duke class, string constants were initiated to improve
code readability and reusability.

syncode98 committed:
Implement error handling for done command

A new method, readDone(), has been defined to handle three
possible errors that might occur when users use the "Done"
command.

Three new print methods have been implemented to print the correct
output for the errors.

A new Illegal Number Exception class has been defined to tackle
an exception that occurs during the readDone() method.

syncode98 committed:
Deal with exceptions when User inputs Todo tasks

A new class has been initialised to deal with blank
description for the commands for the tasks.

The new readTodo() method deals wtih the
exceptions that arises when the user inputs
any Todo tasks.

syncode98 committed:
Implement Class to print methods

A new PrintMethod class contains the
methods that have to be printed out. Improves
code readability.

The printlines() method has been moved from Task class
to  PrintMethod class.

syncode98 committed:
Implement Error Handling for Event and Deadline tasks

The relevant methods have been implemented to ensure that
if the user does not provide the relevant details to the chatbot, then
the chatbot will prompt for the necessary information(for
deadline and events)

A new class , IllegalPropositionWithoutDate, has been implemented to
tackle the scenario where the user does not appropriately enter
the date for the event or deadline tasks.

syncode98 committed:
Final Tests for Level 5

The Actual and Expected files have been updated to
show most of the features that have been implemented.

The initilaiseMike method moved back to the Duke.java class
from the PrintMethod Java class to ensure that only one Scanner
object is created. This prevents errors that occur while doing
regression testing.

Code refactored to follow coding standards

syncode98 committed:
Merge branch 'branch-Level-5'

syncode98 committed:
Creating packages within project

Three new packages have been created.They are the
duke package to contain all the files, exception package
to contain all the exception classes that were created and
the task package to hold the task,event,todo and deadline
classes.

syncode98 committed:
Creating a class to track the tasks

The TaskArray class was created to read all
the commands and assign them to the task array.
Contains the other relevant methods that are required
for reading the commands.

syncode98 committed:
Code refactoring before final commit for this level

TaskArray class has been added to task package.

The regression testing has been updated to include
all the class paths for the packages.

Formatting done for all files.

syncode98 committed:
Ignore certain files

Contributors.md and README.md have been ignored
from revision control

syncode98 committed:
Merge branch 'branch-A-Packages'

syncode98 committed:
Implement A-CodeQuality

Code has been refactored to follow coding
conventions.

syncode98 committed:
Implementing exceptions

Three exception classes have been defined to
tackle the exceptions that arise.

Try and catch functions have been defined to
 deal with exceptions that would arise in methods.

syncode98 committed:
Implement ArrayList in Project File

The Task Array in TaskArray.java
 has been replaced by an ArrayList of tasks.

Method implemented in Duke.java
to delete tasks. Keyword has also been
implemented to identify command "delete".

The numberOfTasks variable in Task.java
has been deleted since it is redundant.
The getters and setters have been deleted
too.

Methods in PrintMethods have been modified to
accomodate the ArrayList.

syncode98 committed:
Combine two methods to read command

The readDone method and deleteTask
methods in TaskArray.java have been combined
since they have a similar structure. The new method
is named readDoneandDelete.

syncode98 committed:
Code refactoring for Level 6

Unnecessary comments have been removed
from all of the class files.

The input.txt file has been updated to test
the delete feature. Regression testing was
successful. The EXPECTED.TXT file has been
updated to accordingly.

syncode98 committed:
Write to an external file

A new method, writeToFile, has been
implemented in TaskArray.java to transfer
the task contents to an external file, data.txt.

syncode98 committed:
Enable application to detect the existence of data.txt file

A method has been implemented in PrintMethod.java class
to check if the data.txt file is available. The method, printContents,
will then show the tasks from a previous session.

If a file is not present, the application informs the user that a
new file will be created in the directory that contains the src folder.
The createFile() method creates the data.txt file for the user.

syncode98 committed:
Implement methods to change content in data.txt file

A method, searchFile has been implemented in TaskArray.java .
It searches the existing file for the specific task and changes the task.
This is useful when the user enters the done command. Hence, the
searchFile method updates the existing file to replace the task which has
been completed.

The readTask method in TaskArray.java now reads all of the
inputs from the todo,event,deadline commands, instantiates the
task and adds the task to the array.

The createFile method in TaskArray.java creates a file if a file is
not present.

The clearFile method in TaskArray. java clears the contents of the file
if the user does not need it.

The printContents  Method has been moved from PrintMethod class
to Duke.java class.

syncode98 committed:
Implement methods to check status of directory and the file

The methods, checkDirectoryStatus and checkFileStatus
checks if the directory and the file exists. If not, the application
creates it for the users.

syncode98 committed:
Regression Testing on File methods

The text-ui-test/data/data.txt files updates automatically according
to inputs.

The ACTUAL.txt file shows the implemented changes.
(This is observed from the EXPECTED.txt file)

syncode98 committed:
Implement Level 7

Code has been refactored to follow
coding standards

syncode98 committed:
Re-Merge with Master branch

This is done to merge with the master branch.

The previous merge was done with a fast forward option.

syncode98 committed:
Merge branch 'branch-Level-6'

syncode98 committed:
Re commit Level 6

This move is to list the commits that are
supposed to be under Level-6-branch,
which was deleted.

Implement ArrayList in Project File
 fd3b08a97a12d86ad0bb5e57449467af46b2a995 [fd3b08a]

Combine two methods to read command
b6dd92e8ffa33f489e2d386d91c888e7bfa35e49 [b6dd92e]

Code refactoring for Level 6
7df933f632edc4d4c76efa55d4d1191573464cc2 [7df933f]

syncode98 committed:
Merge branch 'branch-Level-6'

syncode98 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/task/TaskArray.java
#	text-ui-test/EXPECTED.TXT
#	text-ui-test/input.txt

Retained both changes in Duke.java and TaskArray.java.

Retained Head changes in EXPECTED.TXT and input.txt.

syncode98 committed:
Implement delete function for external file

The searchFile method in TaskArray.java has been
changed to updateFile. It now takes care of
deleting tasks in the external file.

syncode98 committed:
Final Implementation of Level 7

Regression Testing done to verify that
deletion of tasks is successful.

Refactored the code to follow coding conventions.

syncode98 committed:
Implement additional methods to transfer data to data.txt

The format of the  output that is being written to the
 data.txt file has been changed. The Boolean value
is now represented by 0's and 1's.  Complete tasks are assigned
with a value of 1 while incomplete tasks are assigned with a value
of 0.

The application now loads the pre existing tasks from
the data.txt file. The fileToTask method converts the
task in the data.txt file to its respective task object. After
that, the task is added in to the arraylilst.

syncode98 committed:
Code Refactor for  Level 7

The directory and filepaths are now stored as constant
expressions in TaskArray.java.

The fileData method in TaskArray.java reads the contents
of data.txt and stores them as an Arraylist.

The printContents method in Duke.java class
now shows the path in which the directory would be created eventually.

syncode98 committed:
Regression Testing Level 7

Regression Testing was conducted
to ensure the accuracy of the application.

Code refactored in to follow coding conventions.

Mick609 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+package duke.task;
+
+public class Task {
+    private String task;
```
I am a bit confused. Does the variable "task:String" holds the title of the task? If so, please consider renaming it to something like "taskName" or "taskTitle" for clarification. Otherwise, the Task object would have the method getTask() for this title, which sounds like returning the object itself.

Mick609 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+        this(null, false);
+    }
+
+    public String getTask() {
```
The confusion here caused bu the name of the variable name "task: String". The method named "getTask()" in the class "Task" should return the object of the class, which is not the case here.

Mick609 commented on src/main/java/duke/task/TaskArray.java:
```diff
@@ -0,0 +1,367 @@
+
+    }
+
+    public static void fileData(ArrayList<String> filesLines) {
```
It will be more readable if the name of the method is an action.

Mick609 replied:
Think again, maybe what you mean here is the file is the verb. Please ignore this comment.

Mick609 commented on src/main/java/duke/task/TaskArray.java:
```diff
@@ -0,0 +1,367 @@
+        return task;
+    }
+
+    public static void readDoneAndDelete(String command, String keyword, boolean writeFile) {
```
The name of a boolean variable should be in the format of "isX", for example, "isWriteFileDone". Thus, it is better if the bollean argument "writeFile" can follow this format.

Mick609 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,142 @@
+        System.out.println("Hello " + nameOfUser + " !");
+        PrintMethod.printLines();
+        printContents();
+        //PrintMethod.printLines();
```
Please remove the unwanted commented line.

Mick609 commented on src/main/java/duke/PrintMethod.java:
```diff
@@ -0,0 +1,71 @@
+
+
+    public static void printLines() {
+        System.out.println("-------------------------------------------------------------");
```
The name of the method is "printLines". However, the method only prints one line of the dash.

Mick609 reviewed (commented):
The code is clean and easy to read with only a few little naming problems.

syncode98 committed:
Extract closely related code

The code has been refactored to extract out
the methods that perform a specific task.

The Ui Class takes care of User-interactions.
The Parser Class performs the relevant operations on
the String to extract out the task.

The storage class takes care of the loading of the data from the
data.txt file.

The taskList class modifies the ArrayList which tracks all
of the tasks.

syncode98 committed:
Code refactor for TaskList.java

The class now takes care of the relevant methods
for the taskArrayList.

syncode98 committed:
Code refactoring to implement minimal OOP

The required classes have been implemented
at this level. In the Duke class, the relevant
objects are instantiated to perform the relevant actions.

syncode98 committed:
Refactor Parser.java class

Seperate Command classes have been
created for the different commands that can be
given by the user.

A new exception, InvalidCommand, has
been created to deal with case of the user
entering an invalid command. It is thrown
in the AddCommand.java class.

syncode98 committed:
Code Refactor for OOP implmentation

The command classes have been regrouped into
a command package.

The taskFromFile method has been moved
to the Storage class.It was previously
in the Parser class.

syncode98 committed:
Regression testing for OOP

syncode98 committed:
Implement Finding Command Methods

The FindCommand class finds the task
and shows the user the tasks which
have the task.

syncode98 committed:
Code refactoring for find command method

The InvalidTask exception is thrown when the
user gives a invalid task description.This
exception is thrown by the checkTask method
in FindCommand class

syncode98 committed:
Code refactoring

syncode98 committed:
Code refactoring

syncode98 committed:
Implement date & time in Deadline class

The deadline class now has an additional date
variable, which stores the date of the
task in the taskArrayList.

There are minor code refactors in the
other classes.

syncode98 committed:
Implement time and date features in Event class

The toString() method converts the date and time
of the event to string. It is appended to the task
to give the final output.

syncode98 committed:
Add exception to TaskList.java

If the task equal to null, then the method throws
and exception.

syncode98 committed:
Refactor code in AddCommand.java and Storage.java

In the AddCommand.java, the returnEventDeadline() method
returns the deadline or the event object, after instantiating it
with the appropriate time and date variable.It also finds
the date and time of the task given to it.

In the Storage.java class, the updateFile() method
is modified so that the completion status of the task
will be updated accordingly in the data.txt file.

syncode98 committed:
Implement time in Deadline class

The deadline class now tracks the time
of deadline.

syncode98 committed:
Code refactoring to read input from File

The method taskFromFile() in storage.java now reads the date and Time input
from the files and updates it accordingly to the taskArrayList. The parseString
method in AddCommand.java has been changed to accomodate reading of file.

The classes Deadline.java and Event.java have been refactored
to implement the correct toString() method. Now, the time is shown as
HH:mm and the date is shown as dd MMM yyyy.

syncode98 committed:
Allow multiple formats for input of date

The returnEventDeadline method now allows
users to input date by the d-MM-yyyy format.

Minor code refactoring in the Ui.java
class to catch exception.

syncode98 committed:
Merge pull request #1 from syncode98/branch-Level-8

Merge Branch level 8

syncode98 committed:
Level-8

syncode98 committed:
Format data.txt file

syncode98 committed:
Merge remote-tracking branch 'origin/master'

syncode98 committed:
Add JavaDoc to the Deadline and Event classes

syncode98 committed:
Add JavaDoc to AddCommand and Parser classes

syncode98 committed:
Add JavaDoc to Storage class and Tasklist Class

syncode98 committed:
Add JavaDoc for Command.java class and Ui.java class

syncode98 committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	data/data.txt
Accept head changes for FindCommand.java,Parser.java,
Ui.java,TaskList.java.
Accept master changes for AddCommand.java,Event.java,
Deadline.java,data.txt.

syncode98 committed:
Merge pull request #2 from syncode98/branch_A_JavaDoc

Merge Branch A java doc

syncode98 committed:
Necessary changes in data.txt file

syncode98 committed:
Merge branch 'master' into branch-Level-9

syncode98 committed:
Merge pull request #3 from syncode98/branch-Level-9

Merge Branch level 9

syncode98 committed:
Implement Level 9

syncode98 committed:
Revert "Implement Level 9"

This reverts commit 9201814d57b0677dda4e43ef9f3e1b8cdc06291f.

syncode98 committed:
Refactor Command classes into a package

syncode98 committed:
Merge branch 'master' of https://github.com/syncode98/ip

syncode98 committed:
Regression Testing of Project

After the merging of all of the branches, regression
testing has been done to check if program runs
correctly.

syncode98 committed:
Set theme jekyll-theme-architect

syncode98 committed:
First draft of README.md file

syncode98 committed:
Second draft of README.md

syncode98 committed:
Update README.md file

The Command Summary has been added
into the README.md file.
A few images have been added to show the output.

syncode98 committed:
Set theme jekyll-theme-cayman

syncode98 committed:
Refactor code to update the website

The markdown in the _config.yml file is changed to
fix the issue of the table of content not appearing correctly
in the website.

The pictures in the README.md file are removed and replaced with block
line of codes.

syncode98 committed:
Add the title,description and a table of contents for the website

The _config file is configured to add the Title and the description of the Application.

A table of contents, with appropriate hyperlinks, is added in the README.md.

Hyperlinks are also added in the Store tasks section.

syncode98 committed:
Implement more OOP

The execute method now executes the other
methods within the command classes. Previously,
the methods were executed when the command class
was initialised automatically.

The parseInput() method in Parser class
now parses the input given by the user.

syncode98 committed:
Update README.md doc

The correct hyperlinks have been inserted in the README.md doc.
Previously, the hyperlinks redirected the user to
the README.md doc, instead of the website itself.

Furthermore, changed certain single line codes
into block codes.

syncode98 committed:
Implement Clear Feature

A new command has been introduced to the application.
The user needs to enter"clear" to remove all of the existing tasks
in the application and the data file .

syncode98 committed:
Update READMe.md file

Include find feature in description.

Change certain hyperlinks in the document to point to the correct
sections.

syncode98 committed:
Rectify the errors in previous README.md

The format for the event with date,start time and end time was in line
with the header. The format line was pushed to the next line.

syncode98 committed:
Handle additional exceptions within project

The application now handles the scenario if the user
enters the wrong preposition for a task or did not enter
the slash '/' for certain tasks like deadline or events.

syncode98 committed:
Code refactoring

Add relevant JavaDoc to methods.

Remove unwanted class level members and methods.

syncode98 committed:
Code refactoring

Add the exceptions to the AddCommand class and Storage class.

Edit the READMe.md to change the word "files" to "list"

syncode98 committed:
Refactor completion status

The completion status has been changed from X and tick to N and Y
respectively. This is to help users who might not be able to see these
unicode symbols.

syncode98 committed:
Refactor code to remove sadface unicode symbol
